### PR TITLE
Validate filter input, disable 'Apply' button if invalid.

### PR DESCRIPTION
**Pull request by gijskant**
- Validates filter input: checks if gene name, or
  SNP name are not empty and if a genomic region
  conforms to the required format;
- Gives feedback on the genomic region input;
- Disables 'apply' button until the filter input is
  valid;
- Disables default behaviour of pressing 'Enter'
  (submitting the form) and triggers a click event
  on the 'apply' button instead if enabled.
- Fixes dropping of low-dimensional data concepts in
  the data export.

Fixes NAA-63, NAA-70, NAA-75 and part of NAA-74.

### CODE REVIEW

**Comment by gijskant** [comment]
Also fixes NAA-74 and NAA-69 now.

File: web-app/js/datasetExplorer/highDimDialog.js [hunk 1]
```diff
@@ -140,10 +143,10 @@ HighDimensionDialogService = (function(autocompleteInp) {
      * @private
      */
     service.dropOntoVariableSelection = function (data, el, filter) {
-        if (Object.keys(data.details).indexOf(data.dataTypeId) >= 0 ) {
+        //if (Object.keys(data.details).indexOf(data.dataTypeId) >= 0 ) {
```
Comment (JanKanis) [inline web-app/js/datasetExplorer/highDimDialog.js]: What's the purpose of this?

File: web-app/js/datasetExplorer/highDimDialog.js [hunk 2]
```diff
@@ -185,6 +204,109 @@ HighDimensionDialogService = (function(autocompleteInp) {
...
+                        var valid_segment = false;
+                        // validate pattern
+                        var m = service.chromosome_segment_pattern
+                                .exec(_d);
+                        if (m) {
+                            var chromosome = m[1];
+                            var start = m[2];
+                            var end = m[3];
+                            if ((isNaN(chromosome) || (chromosome >= 1 && chromosome <= 22))
+                                    && start <= end) {
```
Comment (JanKanis) [inline web-app/js/datasetExplorer/highDimDialog.js:255]: nit: start <= end or start < end?

Comment (gijskant) [inline web-app/js/datasetExplorer/highDimDialog.js:255]: I guess, selecting exactly one basepair position should be possible...

**Pull request merged**